Add customizable format-spec string for adding title to body

meliache opened the pull request:
Can be specified for each pretty-hydra via `:title-body-format-spec`. Default can be customized via `pretty-hydra-default-title-body-format-spec`.

I use this to customize amount of whitespace and newlines around title. Otherwise, e.g. changing newlines between title and body might be difficult with just `:formatter`. For my original problem that this solves see issue #49 created by me.

My format-spec names are a bit lengthy, I would be open to suggestions how to improve the naming. And if you think this customization can be achieved by other means, feel free to tell me how and close this :)

Solves #49

meliache committed:
Add customizable format-spec string for adding title to body

Can be specified for each pretty-hydra via `:title-body-format-spec`. Default
can be customized via `pretty-hydra-default-title-body-format-spec`.

Solves issue #49.

meliache commented on pretty-hydra.el:
```diff
@@ -218,10 +232,12 @@ See `pretty-hydra-define' and `pretty-hydra-define+'."
-         (docstring (->> heads-plist
-                         (pretty-hydra--gen-body-docstring separator)
-                         (pretty-hydra--maybe-add-title title)
+         (docstring (->> (pretty-hydra--maybe-add-title
```
Here I had to replace a bit of the `->>` threading by old-school sexp-nesting, because I added an optional argument to `pretty-hydra--maybe-add-title`. Therefore `docstring` can't be the last argument to it anymore. I have no experience with `dash.el` threading macros so if you know a prettier way how to solve this, let me know.

jerrypnz replied:
What you can do is changing the parameter list and keep `docstring` the last argument and make the new parameter mandatory (you are passing it to it anyway).

```elisp
(defun pretty-hydra--maybe-add-title (title title-body-format-spec docstring)
```

meliache replied:
Implemented your suggestion in 0f9a68582478e86f9a643f9626ce5f16ecce3468

jerrypnz reviewed (commented):
Thanks for the PR and sorry for the super late response (I haven't been able to spend time on OSS projects lately).

I replied to your comment re threading. Let me know what you think.

meliache committed:
Make docstring last argument of --maybe-add-title again

This commit changes the signature of `pretty-hydra--maybe-add-title`, so that
`title-body-format-spec` is now a  required argument (though it can be `nil`)
and `docstring` is again the last argument. This faciliates threading with
`->>`.

Further, this commit fixes some minor docstring typos which I found thanks to
`jinx.el`

meliache commented:
Sorry for kind of forgetting about this PR, I also got busy shortly after and I changed all my custom hydras to transients instead, which is sufficient if I want automatic formatting of a menu in columns and rows. Still I think this package can be very useful. I went through my old stale org TODO-items, found this PR and decided to implement the one only comment from you. I tested it manually with the example pretty-hydra's from the README.

meliache committed:
Fix docstrings to silence `elisp-flymake-checkdoc` linter

jerrypnz commented:
@meliache Thanks for the update and considering my comment. It looks good to me! To be honest I haven't touched my own Emacs config for ages, not to mention keeping this project up to date. Glad to find it useful for some people though.

jerrypnz commented:
@meliache Thanks for the update and considering my comment. It looks good to me! To be honest I haven't touched my own Emacs config for ages, not to mention keeping this project up to date. Glad to find it useful for some people though.
